### DISCUSSION TITLE
In version 0.62.2, the Chinese name canvas note cannot display a thumbnail

### DISCUSSION THREAD

**Discussion by tiger7651** [TriliumNext/Trilium#4478] [Q&A]
In version 0.62.2, the Chinese name canvas note cannot display a thumbnail;  but in version 0.60.4, it can display a thumbnail of the Chinese name canvas note. Please fix this .
![image](https://github.com/zadam/trilium/assets/150779117/a551c401-62ee-4dac-9a4e-9fb946ffe3d2)

**Comment by Nriver** [comment]
Yeah, I can reproduce it with the following sample note.

[Canvas examples.zip](https://github.com/zadam/trilium/files/13469206/Canvas.examples.zip)

**Reply by Nriver** [reply]
The correct url is like this:
http://127.0.0.1:37840/api/images/kF0NpbbOzBUv/testnote?0.14725540832460204
The wrong one is like this:
http://127.0.0.1:37840/api/images/ivRDSVKMDvP1/?0.5730820288320337

I think this is what goes wrong:
https://github.com/zadam/trilium/blob/3b10bb742b72ad94556326f6145b8983c6c0bfd2/src/public/app/services/content_renderer.js#L120

The Chinese characters are sanitized, so pure Chinese title will left empty string here thus causing an error. Maybe we can change it to urlencode by using `encodeURIComponent`?
I noticed the regex `/[^a-z0-9-.]/gi` is used in several different places. Should we change them too? @zadam

**Reply by zadam** [reply]
Thanks for repro, I tried to fix all the occurrences with `encodeURIComponent` as you suggested.
